openverse_catalog/dags/common/dag_factory.py

```diff
@@ -126,9 +127,17 @@ def _push_output_paths_wrapper(
         ti.xcom_push(key=f"{media_type}_tsv", value=store.output_path)
 
     logger.info("Running provider function")
+
+    start_time = time.perf_counter()
     # Not passing kwargs here because Airflow throws a bunch of stuff in there that none
     # of our provider scripts are expecting.
-    return func(*args)
+    data = func(*args)
+    end_time = time.perf_counter()
+
+    duration = end_time - start_time
+    ti.xcom_push(key="duration", value=duration)
```

Comment:
til about [xcoms](https://airflow.apache.org/docs/apache-airflow/stable/concepts/xcoms.html), really nice feature.

Reply:
They are so useful!! I discuss it a bit more in the issue (#356) - they're great for splitting out more complicated tasks into multiple steps and pass the information between them :smile:

Reply:
I just found [_your comment_ ](https://github.com/WordPress/openverse-catalog/pull/369#discussion_r810315410) which mentions everything I noticed, ha 😅